### ISSUE TITLE
windows: osquery crashing on query of processes table with multibyte values

### ISSUE THREAD

**Issue by poppyseedplehzr** [facebook/osquery#3039]
As of `2.2.0` the Windows Processes virtual table is crashing, potentially due to us not handling multi-byte values correctly. JohnDoe in slack offered up steps to repro the crash, as well as a crash dump file for Windows:
```
PS C:\> .\procdump64.exe -e -h -ma -g -x .\osq_dumps\ ".\build_2.3.4\osqueryi.exe"

ProcDump v8.0 - Writes process dump files
Copyright (C) 2009-2016 Mark Russinovich
Sysinternals - www.sysinternals.com
With contributions from Andrew Richards

Process:               osqueryi.exe (25112)
CPU threshold:         n/a
Performance counter:   n/a
Commit threshold:      n/a
Threshold seconds:     10
Hung window check:     Enabled
Log debug strings:     Disabled
Exception monitor:     Unhandled
Exception filter:      *
Terminate monitor:     Disabled
Cloning type:          Disabled
Concurrent limit:      n/a
Avoid outage:          n/a
Number of dumps:       1
Dump folder:           .\osq_dumps\
Dump filename/mask:    PROCESSNAME_YYMMDD_HHMMSS


Press Ctrl-C to end monitoring without terminating the process.

I0303 18:22:03.771044  4984 options.cpp:61] Verbose logging enabled by config option
I0303 18:22:03.771044  4984 process_ops.cpp:164] Unable to find environment variable (203): TERM
Using a [1mvirtual database[0m. Need help, type '.help'
osquery> select pid from processes limit 1;
[18:22:15] Exception: E06D7363.?AVrange_error@std@@
[18:22:15] Unhandled: E06D7363.?AVrange_error@std@@
[18:22:15] Dump 1 initiated: .\osq_dumps\osqueryi.exe_170303_182215.dmp
[18:22:15] Dump 1 writing: Estimated dump file size is 71 MB.
[18:22:16] Dump 1 complete: 71 MB written in 1.0 seconds
[18:22:16] Dump count reached.
```
However in the `2.1.2` build this behavior doesn't seem to exist:
```
PS C:\> .\procdump64.exe -e -h -ma -g -x .\osq_dumps\ ".\build_2.1.2\osqueryi.exe"

ProcDump v8.0 - Writes process dump files
Copyright (C) 2009-2016 Mark Russinovich
Sysinternals - www.sysinternals.com
With contributions from Andrew Richards

Process:               osqueryi.exe (27968)
CPU threshold:         n/a
Performance counter:   n/a
Commit threshold:      n/a
Threshold seconds:     10
Hung window check:     Enabled
Log debug strings:     Disabled
Exception monitor:     Unhandled
Exception filter:      *
Terminate monitor:     Disabled
Cloning type:          Disabled
Concurrent limit:      n/a
Avoid outage:          n/a
Number of dumps:       1
Dump folder:           .\osq_dumps\
Dump filename/mask:    PROCESSNAME_YYMMDD_HHMMSS


Press Ctrl-C to end monitoring without terminating the process.

I0303 18:30:11.572438  9964 options.cpp:61] Verbose logging enabled by config option
Using a [1mvirtual database[0m. Need help, type '.help'
osquery> select pid from processes limit 1;
+-----+
| pid |
+-----+
| 0   |
+-----+
osquery>
```

**Comment by poppyseedplehzr** [comment]
I suspect [this is the issue](https://github.com/facebook/osquery/pull/2889)

**Comment by stachdude** [comment]
also maybe it's related to problems we experience [in this issue](https://github.com/facebook/osquery/issues/2991)

**Comment by tehmas** [comment]
This error isn't occurring in version 2.3.4.1 .
![capture](https://cloud.githubusercontent.com/assets/9120583/23613407/1a2e4b1c-02a1-11e7-84d5-7ac8eb30a2e4.PNG)

**Comment by tehmas** [comment]
This error isn't occurring in version 2.3.4.
![capture](https://cloud.githubusercontent.com/assets/9120583/23614032/6965e7ec-02a3-11e7-9bce-7695d7188f72.PNG)

**Comment by poppyseedplehzr** [comment]
Not sure this is related to the non-US localization any more. I just tested on Windows 10 with Japanese localization and latest version didn't crash. I have a crash dump that JohnDoe sent me, I'll tear into that sometime today and try to further diagnose

@stachdude not sure it's related to the thrift issues, as this bug existed in Windows 10, and I believe the thrift issue is with how we built thrift having problems on Windows 7, but I haven't yet had time to diagnose that, sorry!

**Comment by poppyseedplehzr** [comment]
It was mentioned this might be induced by having a _native_ non-US locality, and that the locality this was triggering on was ru-RU.  Below is some more specific details about the crash:
```
+        this    0x00007ff60fb8d820 {osqueryi.exe!std::wstring_convert<std::codecvt_utf8_utf16<wchar_t,1114111,1>,wchar_t,std::allocator<wchar_t>,std::allocator<char> > osquery::converter} {...}    std::wstring_convert<std::codecvt_utf8_utf16<wchar_t,1114111,1>,wchar_t,std::allocator<wchar_t>,std::allocator<char> > *
+        _Dnext    0x000002684259dff0 L""    wchar_t *
+        _First    0x0000006e793aea01 "дминистраторы"    const char *
+        _First_sav    0x0000006e793aea00 "Администраторы"    const char *
+        _Last    0x0000006e793aea0e ""    const char *
+        _Wbuf    L""    std::basic_string<wchar_t,std::char_traits<wchar_t>,std::allocator<wchar_t> >
+        _Wstr    L""    std::basic_string<wchar_t,std::char_traits<wchar_t>,std::allocator<wchar_t> >
```

**Comment by poppyseedplehzr** [comment]
This issue was fixed with #3170
